5-appinfra/modules/alloydb-psc-setup/db.tf

```
@@ -76,6 +76,6 @@ resource "google_compute_forwarding_rule" "psc_fwd_rule_consumer" {
 
 resource "google_project_iam_member" "cymbal_bank" {
   project = var.app_project_id
-  role    = "alloydb.admin"
+  role    = "roles/alloydb.admin"
   member  = "serviceAccount:bank-of-anthos@${var.cluster_project_id}.iam.gserviceaccount.com"
```

Same here

---

you mean the service account? "serviceAccount:bank-of-anthos@${var.cluster_project_id}.iam.gserviceaccount.com"

we can put this on a variable and make the user define it

---

renamed the resource name on commit 
b9f4b1b

---

Yup @caetano-colin , the new resource name looks good! But we should also name the service account appropriately to avoid conflict, for example based on the app and service name (will need to be plumbed through.
```suggestion
  member  = "serviceAccount:{var.app_name}-{var.service_name}@${var.cluster_project_id}.iam.gserviceaccount.com"
```

---

If this PR is blocking anything, we can also add a TODO and follow-up in a subsequent PR given it will take some additional changes to plumb.

---

I don't think this PR is a blocker, so we can do this

There is a catch: service accounts names also have a max length of 30 characters, so we might want to use app/service acronyms

---

Regarding the above mentioned issue, would that be reasonable:

#### Apps Acronym

Max length for app acronym would be 8 characters.

| Application   | Acronym |
|---------------|---------|
| Cymbal Bank   | cym-bank      |

#### Micro Services Acronyms

Max length for a service acronym would be 12 characters.

| Service Name             | Acronym   |
|--------------------------|-----------|
| Accounts Contacts        | act-contacts  |
| Accounts User Service    | act-usersvc  |
| Frontend                 | frontend     |
| Ledger Balancereader     |   ldgr-breader        |
| Ledger Ledgerwriter      |   ldgr-lwriter       |
| Ledger Transaction History | ldgr-history        |

then we can also use this convention for project ids, would leave us 10 characters for prefixes/suffixes or other identifiers

---

actually I just found out that alloydb is supported for IAM principal using Worload Identity in GKE, so we can assign this role without using a GCP service account and reference the kubernetes service account directly using this principal:

```
principal://iam.googleapis.com/projects/PROJECT_NUMBER/locations/global/workloadIdentityPools/PROJECT_ID.svc.id.goog/subject/ns/NAMESPACE/sa/SERVICEACCOUNT
```

---

not sure if that would require the k8s sa to exist before the assignment